include/boost/geometry/algorithms/detail/equals/point_point.hpp

```diff
@@ -36,17 +33,16 @@ namespace detail { namespace equals
     \brief Internal utility function to detect of points are disjoint
     \note To avoid circular references
  */
-template <typename Point1, typename Point2>
-inline bool equals_point_point(Point1 const& point1, Point2 const& point2)
+template <typename Point1, typename Point2, typename Strategy>
+inline bool equals_point_point(Point1 const& point1, Point2 const& point2,
+                               Strategy const& )
 {
-    return ! detail::disjoint::disjoint_point_point(point1, point2);
+    return Strategy::apply(point1, point2);
```

Comment:
Why we need a special strategy for checking if 2 points are equal? Is it the same for all CS? Unless if you consider normalization, but in this case how normalization strategy is passes here? What am I missing...

Reply:
In general yes, this could be done the same for all coordinate systems. For this we'd have to agree what's the valid range of coordinates, e.g. if `lon_deg = -180+eps/2` is valid, since this would be true: ``bg::math::equals(lon_deg + 360, 180), this coordinate would be ambiguous and cartesian-like comparison would fail. Currently we have this normalization in place which even if the user represents the coordinates ambiguously (`-180` vs `180` or even as something outside valid range) the relation operations still works. In order for this to work we need separate strategies for various CSes even for points.

Reply:
I do not have a strong opinion here, since I cannot predict the user needs, but what about fixing a range e.g. `(-180,180]`, then if the user compare invalid coordinates the result is "unexpected" as with invalid geometries.

Reply:
In any case, this should be done in a new PR.

Reply:
Yes, the normalization could also be a part of `correct`. However I think that assuming that -180 is correct is common practice and if we changed the existing behavior users could get unexpected results.